cli/main.c

```diff
@@ -157,33 +144,65 @@ int main(int argc, char** argv) {
   // FIXME: this is temp till arg parse implemented.
   PkResult result;
 
-  if (argc == 1) {
+  static const char* const usage[] = {
+    "pocket [option] ... [-c cmd | file | -] [arg] ...",
+    NULL,
+  };
+
+  int debug = 0;
+  int quiet = 0;
+  int version = 0;
+  const char* cmd = NULL;
+
+  struct argparse_option flags[] = {
+    OPT_HELP(),
+    OPT_GROUP("Basic options"),
+    OPT_STRING('c', "cmd", &cmd, "Evaluate and run the passed string."),
+    OPT_BOOLEAN('d', "debug", &debug, "Compile and run a debug version."),
+    OPT_BOOLEAN('q', "quiet", &quiet,
+               "Don't print version and copyright statement on REPL startup."),
+    OPT_BOOLEAN('v', "version", &version,
+               "Prints the pocketlang version number and exit."),
+    OPT_END(),
+  };
+
+  struct argparse argparse;
+  argparse_init(&argparse, flags, usage, 0);
+  argparse_describe(&argparse,
+        "\nPocketlang is a small and fast functional language written in C.",
+        "\nCopyright(c) 2020 - 2021 ThakeeNathees." \
+        "\nFree and open source software under the terms of the MIT license.");
+  argc = argparse_parse(&argparse, argc, argv);
+
+  if (argc == 0) {
     options.repl_mode = true;
     repl(vm, &options);
-
-  } if (argc >= 3 && strcmp(argv[1], "-c") == 0) {
-
-    PkStringPtr source = { argv[2], NULL, NULL };
-    PkStringPtr path = { "$(Source)", NULL, NULL };
-
-    result = pkInterpretSource(vm, source, path, NULL);
-    pkFreeVM(vm);
-    return result;
   }
 
-  PkStringPtr resolved = resolvePath(vm, ".", argv[1]);
-  PkStringPtr source = loadScript(vm, resolved.string);
+  if (argc > 0) {
+    if (debug != 0) {
+      options.debug = true;
```

Comment:
the problem here is that if the `debug == 0` `options.debug` will be un-initialized which cause UB (but in debug mode it'll be initialized to `false`).

Reply:
@xSavitar `options.debug` was never set to false, if the `debug` value is false. And it's un initialized.